src/features/vote/RevealPhase.tsx

```diff
@@ -0,0 +1,290 @@
+                  </div>
+                </td>
+                <td>
+                  <div>
+                    <p className="vote">{el.vote}</p>
+                  </div>
+                </td>
+                <td>
+                  <div>
+                    {el.revealed
+                      ? "Revealed"
+                      : el.vote !== UNDEFINED_VOTE && !el.revealed
+                      ? "Reveal"
+                      : el.vote === UNDEFINED_VOTE
+                      ? "Uncommitted"
+                      : null}
+                  </div>
+                </td>
+              </tr>
+            );
+          })}
+        </tbody>
+      </table>
+      <div className="end-row">
+        <div className="end-row-item">
+          {round.snapshotId === "0" ? (
+            <Button
+              onClick={() => {
+                if (!signer || !votingContract || !provider) return;
+                votingContract.functions["snapshotMessageHash"]().then(
+                  (hash) => {
+                    const sigHash = hash[0];
+                    if ((window as any).ethereum) {
+                      const mm = (window as any).ethereum;
+                      const Web3 = new web3(mm);
+
+                      // Make sure we use the hot address if the are using a two key contract.
+                      let va = votingAddress;
+                      if (hotAddress) va = hotAddress;
+                      if (va) {
+                        getMessageSignatureMetamask(Web3, sigHash, va).then(
+                          (msg) => {
+                            snapshotCurrentRound(votingContract, msg).then(
+                              (tx) => {
+                                // TODO: Refetch state after snapshot.
+                                console.log("success?", tx);
+                              }
+                            );
+                          }
+                        );
+                      }
+                    }
+                  }
+                );
+              }}
+              variant="secondary"
+            >
+              {signer ? "Snapshot Round" : "Connect Wallet to Snapshot"}
+            </Button>
+          ) : canReveal ? (
+            <Button
+              type="button"
+              onClick={() => {
+                // WIP. Comment out for now.
+                // console.log("encryptedVotes", encryptedVotes);
+                if (encryptedVotes.length && activeRequests.length) {
+                  const postData = [] as PostRevealData[];
+                  activeRequests.forEach((el, index) => {
+                    const datum = {} as PostRevealData;
+                    // I believe latest events are on bottom. requires testing.
+                    const latestVotesFirst = [...encryptedVotes].reverse();
+                    const findVote = latestVotesFirst.find(
+                      (x) => x.identifier === el.identifier
```

Comment:
do we not need to match against more fields here?

Reply:
Good call, missed this one.